metricbeat/module/postgresql/replication/data.go

```diff
@@ -0,0 +1,53 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package replication
+
+import (
+	"time"
+
+	s "github.com/elastic/beats/v7/libbeat/common/schema"
+	c "github.com/elastic/beats/v7/libbeat/common/schema/mapstrstr"
+)
+
+// Based on https://www.postgresql.org/docs/9.2/static/monitoring-stats.html#PG-STAT-DATABASE-VIEW
+var schema = s.Schema{
+	"pid": c.Int("pid"),
+	"user": s.Object{
+		"id":   c.Int("usesysid", s.Optional),
+		"name": c.Str("usename"),
+	},
+	"application_name": c.Str("application_name"),
+	"client": s.Object{
+		"address":  c.Str("client_addr"),
+		"hostname": c.Str("client_hostname"),
+		"port":     c.Int("client_port", s.Optional),
+	},
+	"backend_start": c.Time(time.RFC3339Nano, "backend_start"),
+	"backend_xmin":  c.Time(time.RFC3339Nano, "backend_xmin", s.Optional),
```

Comment:
The backend_xmin should not be of type Time. This is 32 bit quantile, so integer perhaps ?

Reply:
++ I also think so. From the definition, it seems it's a transaction ID.

Based on the above and [this](https://github.com/cube-js/cube/blob/master/rust/cubesql/cubesql/src/compile/engine/information_schema/postgres/pg_stat_activity.rs#L37) I think it should ideally use unsigned int.

@ishleenk17 It seems in `mapstr` package i.e., `c.Uint` isn't there and just `c.Int` is there that's using `toInt` function.

- https://pkg.go.dev/strconv#ParseUint
- https://github.com/elastic/beats/blob/main/libbeat/common/schema/mapstrstr/mapstrstr.go#L112
- https://github.com/elastic/beats/blob/main/libbeat/common/schema/mapstrstr/mapstrstr.go#L118

@ishleenk17 What do you think we should do in this case? Should we add a new function `Uint` in `mapstr` package? If yes, should we have it in the same PR or a separate PR for it?

Reply:
@zxcSora You used `c.Int`. You might have added new functions to mapstr package if we are going to use uint32. If the correct data type is not used, values might overflow and result in incorrect metrics.